diff --git a/main.go b/main.go
index 638af00..af15140 100644
--- a/main.go
+++ b/main.go
@@ -2,6 +2,7 @@ package main
 
 import (
 	"log"
+	"time"
 )
 
 func producer(strings []string) (<-chan string, error) {
@@ -10,6 +11,7 @@ func producer(strings []string) (<-chan string, error) {
 	go func() {
 		defer close(outChannel)
 		for _, s := range strings {
+			time.Sleep(time.Second * 3)
 			outChannel <- s
 		}
 	}()
