diff --git a/main.go b/main.go
index 08d7580..c45418b 100644
--- a/main.go
+++ b/main.go
@@ -31,6 +31,7 @@ func sink(ctx context.Context, values <-chan string) {
 			log.Print(ctx.Err().Error())
 			return
 		case val, ok := <-values:
+			log.Println(val)
 			if ok {
 				log.Printf("sink: %s", val)
 			}
@@ -44,10 +45,10 @@ func main() {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	go func() {
-		time.Sleep(time.Second * 5)
-		cancel()
-	}()
+	// go func() {
+	// 	time.Sleep(time.Second * 5)
+	// 	cancel()
+	// }()
 
 	outputChannel, err := producer(ctx, source)
 	if err != nil {
